File: app/src/main/java/com/example/tree/Main/DataModels_Main.java
package com.example.tree.Main;


import com.google.gson.annotations.SerializedName;

import java.lang.reflect.Member;
import java.time.LocalDate;
import java.util.Map;

public class DataModels_Main {
    @SerializedName("distance")
    private double distance;
    @SerializedName("transport")
    private int transport;

    @SerializedName("treeLevel")
    public int treeLevel;

    @SerializedName("treeCount")
    public int treeCount;

    @SerializedName("carbonMap")
    public Map<LocalDate, Double> carbonMap;

    public DataModels_Main(double distance, int transport){
        this.distance=distance;
        this.transport=transport;
    }
    public double getDistance() {
        return distance;
    }

    public int getTransport() {
        return transport;
    }


    public int getTreeLevel() {
        return treeLevel;
    }

    public int getTreeCount() {
        return treeCount;
    }

    public Map<LocalDate, Double> getCarbonMap() {
        return carbonMap;
    }

    public void setTreeCount(int treeCount) {
        this.treeCount = treeCount;
    }

    public void setTreeLevel(int treeLevel) {
        this.treeLevel = treeLevel;
    }

    public void setCarbonMap(Map<LocalDate, Double> carbonMap) {
        this.carbonMap = carbonMap;
    }
}

File: app/src/main/java/com/example/tree/Main/DataModels_Main.kt
package com.example.tree.Main

import com.example.tree.neighbor.GetModel_list
import com.google.gson.annotations.SerializedName
import java.io.Serializable


class Data_Request(
    val distance: Double,
    val transport: Int
):Serializable


data class Data_Response(
    var treeLevel : Int,
    var treeCount : Int,
    var totalReduction : Double,
    var levelReduction : Double
):Serializable

data class Get_Response(
    var tree : String,
    var info : String



):Serializable

data class GetModel_alllist(
    var memberId : Long,
    var treeCount: Int,
    var name: String
)

class getAllList(
    @SerializedName("allList")
    val getallList : List<GetModel_alllist>

)